src/server/server.ts

```diff
@@ -382,31 +377,25 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
    *
    * @param file - uploading file
    */
-  private async parseFileDataIfReady(file: UploadedFile): Promise <void | ApiResponse['payload']> {
-
+  private async checkFileFullness(file: UploadingFile): Promise <void | ApiResponse['payload']> {
     /**
      * Check is file fully uploaded
      */
-    if (file.file.filter(Boolean).length/file.chunks! == 1){
-      let fileData = Buffer.alloc(0);
-
-      /**
-       * Uniting of incoming file chunks
-       */
-      for (let chunk of file.file) {
-        fileData = Buffer.concat([fileData, chunk]);
-      }
-
+    if (file.chunks === file.uploadedChunks) {
```

Comment:
the `uploadedChunks` stores chunks or its length?

Looks like there should be something like

```
file.uploadedChunks.length

```

or 

```
file.uploadedChunksCount
```